### DISCUSSION TITLE
❓ vad bad performance on child speech?

### DISCUSSION THREAD

**Discussion by garymmi** [snakers4/silero-vad#52] [Q&A] [answered]
I have a 16k audio file of child speech
after vad, I got 
start=0.875, dur=1.375
as picture
![image](https://user-images.githubusercontent.com/56832974/113264255-57144d80-9305-11eb-87d0-5ab4f3deff6e.png)
I used model='silero_vad' , call the api
speech_timestamps = get_speech_ts(wav_data, model, num_steps=4)

here is the audio file
https://drive.google.com/file/d/1yivG8OE77TyfJE_KL2IYl-Ltm5cFujnX/view?usp=sharing

could you help me to see what'wrong or teach me how to fine tune the parameters ?
thanks

**Comment by snakers4** [comment, marked as answer]
Several remarks here

- We do not have a lot of children's speech per se in the training data
- This sample just looks too short to be cut by a VAD
- We were planning to release an adaptive post-processing tweak that would make parameter tuning obsolete
- You can use this [snippet ](https://github.com/snakers4/silero-vad#vad-parameter-fine-tuning)to tune params, albeit it should be done on longer audio files, becase this one is too short to be meaninful imo

```
speech_timestamps = get_speech_ts(wav, model,
                                  num_samples_per_window=2000,
                                  num_steps=8,
                                  visualize_probs=True)
```

This snippet produces this image
![image](https://user-images.githubusercontent.com/12515440/113268265-dee76100-92df-11eb-8754-ce0e459d0cf8.png)

**Reply by snakers4** [reply]
Also tuning this maye be a good idea

```
min_speech_samples - minimum speech chunk duration in samples
min_silence_samples - minimum silence duration in samples between to separate speech chunks
```

**Reply by garymmi** [reply]
I got it, thank you very much
